File: 49-GroupAnagrams/49-GroupAnagrams.py
# Last updated: 5/30/2025, 7:35:10 PM
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anag = {}
        res = []
        for word in strs:
            sWord = str(sorted(word))
            if sWord in anag:
                anag[sWord].append(word)
            else:
                anag[sWord] = [word]
        for lis in anag.values():
            res.append(lis)
        return res

File: 49-GroupAnagrams/49-GroupAnagrams.java
// Last updated: 5/30/2025, 7:35:15 PM
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        List<List<String>> listOfLists = new ArrayList<>();

        for (String s : strs) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sortedString = new String(charArray);
            
            map.computeIfAbsent(sortedString, k -> new ArrayList<>()).add(s);
        }

        listOfLists.addAll(map.values());
        
        return listOfLists;
    }

}
